onnxruntime/python/tools/transformers/models/phi2/convert_to_onnx.py

```diff
@@ -0,0 +1,613 @@
+            dtype=torch.int64,
+            device=self.device,
+        )
+        self.get_phi2_torch_model()
+        torch_inputs = self.phi_model.prepare_inputs_for_generation(
+            input_ids, past_key_values=self.phi_model(input_ids, use_cache=True)["past_key_values"]
+        )
+        return torch_inputs["input_ids"], torch_inputs["attention_mask"], torch_inputs["past_key_values"]
+
+    def dynamo_export(self, onnx_path: str):
+        input_ids, attention_mask, past_key_values = self.get_phi2_torch_inputs(self.batch_size, self.sequence_length)
+        self.phi_model(input_ids, attention_mask=attention_mask, past_key_values=past_key_values)
+
+        from torch._dynamo import config
+
+        config.capture_scalar_outputs = True
+
+        logging.info("Exporting Phi2 torch model to ONNX...")
+        torch.onnx.dynamo_export(
+            self.phi_model,
+            input_ids,
+            attention_mask=attention_mask,
+            past_key_values=past_key_values,
+            export_options=torch.onnx.ExportOptions(dynamic_shapes=True),
+        ).save(onnx_path)
+        onnx.checker.check_model(onnx_path)
+        onnx.shape_inference.infer_shapes_path(onnx_path)
+
+    def preprocess_onnx(self, onnx_path_in: str, onnx_path_out: str, func_name: str):
+        model = onnx.load_model(onnx_path_in, load_external_data=True)
+        function_name = None
+        for func in model.functions:
+            if func.name.endswith(func_name):
+                function_name = func.name
+                break
+        assert function_name is not None
+        model = self.unroll_function(model, function_name)
+        model = self.update_edges(model, self.phi2_edge_dict)
+        model = self.simplify_phi2_op_type(model)
+        model = self.remove_dropout_layer(model)
+        onnx.save_model(
+            model,
+            onnx_path_out,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+        )
+
+    def optimize_phi2_onnx(self, onnx_path: str, onnx_path_opt: str):
+        import uuid
+
+        from fusion_options import FusionOptions
+        from optimizer import optimize_model
+
+        processed_onnx_path = f"{uuid.uuid1()}.onnx"
+        model = onnx.load_model(onnx_path, load_external_data=True)
+        model = self.process_graph_io(self.phi_config, model)
+        onnx.save_model(
+            model,
+            processed_onnx_path,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+            location=processed_onnx_path + ".data",
+        )
+
+        optimization_options = FusionOptions("phi")
+        optimizer = optimize_model(
+            processed_onnx_path,
+            model_type="phi",
+            num_heads=self.phi_config.num_attention_heads,
+            hidden_size=self.phi_config.hidden_size,
+            opt_level=0,
+            optimization_options=optimization_options,
+            only_onnxruntime=False,
+        )
+
```

Comment:
Maybe do some operator statistics, and do sanity check here (like expected number of GQA and Attention).